packages/alfa-rules/src/sia-r70/rule.ts

```diff
@@ -0,0 +1,140 @@
+import { Rule, Diagnostic } from "@siteimprove/alfa-act";
+import { Document, Element, Namespace } from "@siteimprove/alfa-dom";
+import { Predicate } from "@siteimprove/alfa-predicate";
+import { Err, Ok } from "@siteimprove/alfa-result";
+import { Page } from "@siteimprove/alfa-web";
+import { List } from "@siteimprove/alfa-list";
+import { Iterable } from "@siteimprove/alfa-iterable";
+
+import { expectation } from "../common/expectation";
+import { hasChild, isRendered, isDocumentElement } from "../common/predicate";
+
+const { isElement, hasName, hasNamespace } = Element;
+const { and, test } = Predicate;
+
+const deprecated = [
+  "applet",
+  "basefont",
+  "bgsound",
+  "big",
+  "blink",
+  "center",
+  "content",
+  "dir",
+  "font",
+  "frame",
+  "frameset",
+  "hgroup",
+  "image",
+  "keygen",
+  "marquee",
+  "menuitem",
+  "nobr",
+  "noembed",
+  "noframes",
+  "plaintext",
+  "rb",
+  "rtc",
+  "shadow",
+  "spacer",
+  "strike",
+  "tt",
+  "xmp",
+];
+
+export default Rule.Atomic.of<Page, Document>({
+  uri: "https://alfa.siteimprove.com/rules/sia-r70",
+  evaluate({ device, document }) {
+    return {
+      applicability() {
+        return test(hasChild(isDocumentElement), document) ? [document] : [];
+      },
+
+      expectations(target) {
+        const deprecatedElements = target
+          .descendants({ flattened: true, nested: true })
+          .filter(isElement)
+          .filter(
+            and(
+              hasNamespace(Namespace.HTML),
+              hasName("acronym", ...deprecated),
```

Comment:
Is there a reason for `"acronym"` being spelled out on its own?

Reply:
hasName (https://github.com/Siteimprove/alfa/blob/01db7c61462ef8bdd0badcb2405119088acd463d/packages/alfa-dom/src/node/element/predicate/has-name.ts#L20-L22) asks for `string `and `array<String>`

Reply:
Ah, `deprecated` should be asserted `const` to fix that:

```ts
const deprecated = [
  "acronym",
  ...
] as const;
```

`hasName()` indeed requires _at least 1_ name, so TypeScript needs to know that the list of names has more than one value. A `const` assertion will make TypeScript infer the type of the array as a fixed-size tuple `[string, string, ...]` rather than `string[]`.